### DISCUSSION TITLE
Trying to open usdview shipped with maya-usd 0.9 installation.

### DISCUSSION THREAD

**Discussion by ChokmahBinah** [Autodesk/maya-usd#1378] [Q&A]
**Is your feature request related to a problem? Please describe.**
Hello, I am trying to open usdview using this file cntaining a character of mine :

![Astronaut_mayaUSD](https://user-images.githubusercontent.com/33164632/116440512-e0f5fe80-a850-11eb-984e-5cdd5ac07bee.png)

Is there a way to open the usdview bin shipped with maya-usd 0.9 installation. ?

Thanks in advance :)

**Describe the solution you'd like**
An easy way (if possible) to open usdview.

**Describe alternatives you've considered**

**Additional context**

**Comment by seando-adsk** [comment]
### Using UsdView with Maya 2022
You can access UsdView with an installation of the MayaUSD plug-in and Maya 2022. It can be accessed in two ways:

1. Run UsdView from within Maya 2022
2. Run UsdView from the command line
 
**1. To run UsdView from within Maya 2022**
Run the following python script from the **Script Editor**:

```
import maya.cmds as cmds
import ufe
from pxr import Usd
import mayaUsd as uLib
import os
import sys
import subprocess
if ufe.GlobalSelection.get().empty():
    cmds.error('Select any ufe item under (and including) gateway node')
ufeItem = ufe.GlobalSelection.get().back()
ufePathGateway = ufe.Path(ufeItem.path().segments[0])
stage = uLib.ufe.getStage(str(ufePathGateway))
sdfLayer = stage.GetRootLayer()
mayaVer = int(cmds.about(q=True, majorVersion=True))
if mayaVer == 2022:
	mayapy = os.path.join(os.environ['MAYA_LOCATION'], 'bin', 'mayapy{ver}'.format(ver='' if sys.version_info.major == 3 else '2'))
else:
	mayapy = os.path.join(os.environ['MAYA_LOCATION'], 'bin', 'mayapy')
usdViewPath = os.path.join(os.environ['USD_LOCATION'], 'bin', 'usdview')
# Install OpenGL module, if needed
try:
    import OpenGL
except:
    subprocess.check_call([mayapy, '-m', 'pip', 'install', 'PyOpenGL==3.1.0'])
CREATE_NO_WINDOW = 0x08000000
subprocess.Popen([mayapy, usdViewPath, sdfLayer.realPath], creationflags=CREATE_NO_WINDOW)
```
**Tip**: For quick access, you can also add UsdView to a shelf button in Maya by selecting and dragging the text to the shelf.

**2. To run UsdView from the Command Line**
**Important**: The following instructions are specific to using the Windows operating system. For other OS platforms (ie. Mac and Linux) paths must be adjusted accordingly.
**Note**: The version of MayaUSD shipped with Maya 2022 was built using python from Maya and `mayapy`, so UsdView will run with `mayapy`. To run with Python 2, use `mayapy2` (instead of "mayapy") and change to `USD3` (from “USD2”).
Open a **command prompt** and run the following commands, replacing the _<version>_ tags with your appropriate version:
```
set PATH=C:\Program Files\Autodesk\Maya<VersionNumber>\bin;C:\Program Files\Autodesk\MayaUSD\Maya<VersionNumber>\<plugin-version>\mayausd\USD3\bin;C:\Program Files\Autodesk\MayaUSD\Maya<VersionNumber>\<plugin-version>\mayausd\USD3\lib;%PATH%
mayapy -m pip install PyOpenGL==3.1.0
mayapy C:\Program Files\Autodesk\MayaUSD\Maya<VersionNumber>\<plugin-version>\mayausd\USD3\bin\usdview <your_file>
```
**Compatibility Note**: UsdView is accessible for Maya 2020. However, Maya 2020 exclusively supports python2. The aforementioned scripts would have to be adjusted accordingly for py2. Replace "USD3" with just "USD" in the scripts for use with Maya 2020.

**Comment by seando-adsk** [comment]
Hi @ChokmahBinah, as you can see above I posted instructions for how to run usdview. 

I also converted this issue into a discussion as we would also like to understand why you would like to run usdview? Is there something missing from within Maya? For everyone else, please feel free to add your reason for needing usdview to this discussion.

**Reply by ChokmahBinah** [reply]
Hello @seando-adsk ,
Thanks you a lot for your help !

I am testing USD in my spare time being mainly a Maya user. 
But to better understand USD and go deeper for developing tools I think it's interesing to get this kind of infos from usdview.

![image](https://user-images.githubusercontent.com/33164632/117198465-59c4ff80-ade9-11eb-9e56-ee1cb3f7d167.png)

I sometimes use Houdini too and there I can see this informations easly too :

![image](https://user-images.githubusercontent.com/33164632/117198661-942e9c80-ade9-11eb-9c6e-64f3c6e76ae0.png)

I would love to be able to see these infos directly in Maya.

Despite that, the Maya integration is pretty cool. Especilally about performances :
This scene example is evaluated at :
70 FPS in Maya Viewport 2.0
30 FPS in Houdini GL in Solaris Context. 

and at :

200 FPS in Maya Hydra Viewport
30  FPS in Houdini Storm in Solaris context 

![image](https://user-images.githubusercontent.com/33164632/117198816-c2ac7780-ade9-11eb-8d48-76f78d6fadab.png)

I wondered if there are plans to support USD layering with Bifrost in the future. (I will understand if you can't tell too much about this ^^)

Thank you again @seando-adsk  :)
